crates/air/build.rs

```diff
@@ -21,5 +21,45 @@ fn main() {
                 MIN_STACK_SIZE, rust_min_stack
             );
         }
+
+        macro_rules! check_feature_enabled {
+                ($($feature:literal),*) => {
+                    #[cfg(not(any($(feature = $feature),*)))]
+                    compile_error!(concat!(
+                        "At least one feature must be enabled: ",
+                        $(concat!("`", $feature, "`, ")),*
+                    ));
+                };
+            }
+
+        macro_rules! check_feature_conflict {
+                ($feature:literal, $($all_features:expr),*) => {
+                    $(
+                        #[cfg(all(feature = $feature, feature = $all_features))]
+                        compile_error!(concat!(
+                            "Conflicting features detected: `", $feature, "` and `", $all_features, "` cannot be enabled together."
+                        ));
+                    )*
+                };
+            }
+
+        #[rustfmt::skip]
+        mod check_layout {
+            check_feature_enabled!("dex", "recursive", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("dex", "recursive", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("recursive", "dex", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("recursive_with_poseidon", "dex", "recursive", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("small", "dex", "recursive", "recursive_with_poseidon", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("starknet", "dex", "recursive", "recursive_with_poseidon", "small", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("starknet_with_keccak", "dex", "recursive", "recursive_with_poseidon", "small", "starknet", "dynamic");
+            check_feature_conflict!("dynamic", "dex", "recursive", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak");
```

Comment:
Pretty verbose and hard to maintain. But maybe there is no other way to express this.

Reply:
There is much better way actually